js/state/siteCache.js

```diff
@@ -0,0 +1,90 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+const Immutable = require('immutable')
+const siteUtil = require('./siteUtil')
+const UrlUtil = require('../lib/urlutil')
+
+const createLocationSiteKeysCache = (state) => {
```

Comment:
In a follow up after this is merged, please move this to `app/common/state`.  Basically no files should be in js/* anymore and we're migrating away from that.